File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    w = []
    h = []
    for i in range(len(sizes)):
        if sizes[i][0] > sizes[i][1]:
            w.append(sizes[i][0])
            h.append(sizes[i][1])
        else:
            w.append(sizes[i][1])
            h.append(sizes[i][0])

    return max(w) * max(h)

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int s = sizes.length;
        List<Integer> w = new ArrayList<>();
        List<Integer> h = new ArrayList<>();
        
        for(int i = 0; i < s; i++) {
            w.add(Math.max(sizes[i][0], sizes[i][1]));
            h.add(Math.min(sizes[i][0], sizes[i][1]));
        }
        
        Collections.sort(w, Collections.reverseOrder());
        Collections.sort(h, Collections.reverseOrder());
        
        int ww = w.get(0);
        int hh = h.get(0);
        
        return ww * hh;
    }
}
